### PR TITLE
migrate memberes from deprecated structs

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

Stackframe is deprecated and will hopefully soon go away

#### Summary of Changes

Move some members out and into invoke context

Fixes #

### CODE REVIEW

File: program-runtime/src/invoke_context.rs [hunk 1]
```diff
@@ -1085,57 +1089,61 @@ impl<'a> InvokeContext<'a> {
...
             .map(|orig_account_lengths| orig_account_lengths.as_slice())
+            .ok_or(InstructionError::CallDepth)
     }
 
+    // Set should alignment be enforced during user pointer translation
     pub fn set_check_aligned(&mut self, check_aligned: bool) {
         self.check_aligned = check_aligned;
     }
 
+    // Sshould alignment be enforced during user pointer translation
```
Comment (Lichtso) [inline program-runtime/src/invoke_context.rs:1113]: typo

File: sdk/src/transaction_context.rs [hunk 2]
```diff
@@ -237,6 +271,26 @@ impl TransactionContext {
     pub fn get_instruction_trace(&self) -> &InstructionTrace {
         &self.instruction_trace
     }
+
```
Comment (Lichtso) [inline sdk/src/transaction_context.rs:274]: left over trial?

File: sdk/src/transaction_context.rs [hunk 3]
```diff
@@ -138,6 +146,21 @@ impl TransactionContext {
         Ok(instruction_context)
     }
 
+    /// Gets an InstructionContext by its nesting level in the stack
+    pub fn get_instruction_context_mut_at(
```
Comment (Lichtso) [inline sdk/src/transaction_context.rs:150]: What do we need the mutable version for? Is it used somewhere?

**Comment by jackcmay** [comment]
@Lichtso https://github.com/solana-labs/solana/pull/24299

**Pull request merged**